### ISSUE TITLE
NPE - Attempt to invoke 'boolean java.io.File.mkdir()' on a null object reference

### ISSUE THREAD

**Issue by barbeau** [OneBusAway/onebusaway-android#440]
From Android developer console, for v2.0.8 beta:

~~~
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.joulespersecond.seattlebusbot/org.onebusaway.android.ui.HomeActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.io.File.mkdir()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2520)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2594)
	at android.app.ActivityThread.access$800(ActivityThread.java:177)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1469)
	at android.os.Handler.dispatchMessage(Handler.java:111)
	at android.os.Looper.loop(Looper.java:194)
	at android.app.ActivityThread.main(ActivityThread.java:5616)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.io.File.mkdir()' on a null object reference
	at com.google.maps.api.android.lib6.gmm6.m.ad.a(Unknown Source)
	at com.google.maps.api.android.lib6.gmm6.c.h.a(Unknown Source)
	at com.google.maps.api.android.lib6.gmm6.c.y.a(Unknown Source)
	at com.google.maps.api.android.lib6.e.bd.a(Unknown Source)
	at com.google.maps.api.android.lib6.e.ev.a(Unknown Source)
	at com.google.maps.api.android.lib6.e.z.a(Unknown Source)
	at com.google.maps.api.android.lib6.e.y.a(Unknown Source)
	at com.google.android.gms.maps.internal.u.onTransact(SourceFile:107)
	at android.os.Binder.transact(Binder.java:385)
	at com.google.android.gms.maps.internal.IMapFragmentDelegate$zza$zza.onCreateView(Unknown Source)
	at com.google.android.gms.maps.SupportMapFragment$zza.onCreateView(Unknown Source)
	at com.google.android.gms.dynamic.zza$4.zzb(Unknown Source)
	at com.google.android.gms.dynamic.zza.zza(Unknown Source)
	at com.google.android.gms.dynamic.zza.onCreateView(Unknown Source)
	at com.google.android.gms.maps.SupportMapFragment.onCreateView(Unknown Source)
	at org.onebusaway.android.map.googlemapsv2.BaseMapFragment.onCreateView(BaseMapFragment.java:184)
	at android.support.v4.app.Fragment.performCreateView(Fragment.java:1965)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1078)
	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1259)
	at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:738)
	at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1624)
	at android.support.v4.app.FragmentController.execPendingActions(FragmentController.java:330)
	at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:547)
	at org.onebusaway.android.ui.HomeActivity.onStart(HomeActivity.java:287)
	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1244)
	at android.app.Activity.performStart(Activity.java:6108)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2477)
	... 10 more
~~~

Device info:

~~~
PURE XL (BLU_PURE_XL) 1
Manufacturer Symphony Teleca 
Android version Android 5.1 
RAM (MB) 3072 
Screen size 1440 × 2560 
Screen density (dpi) 640 
OpenGL ES version 3.0 
Native platform armeabi-v7a 
CPU make Mediatek 
CPU model MT6795
~~~

This is a bug in Google Play Services - more info at https://code.google.com/p/gmaps-api-issues/issues/detail?id=9021.

Google has fixed the issue internally (Jan 20, 2016) but hasn't rolled out the update to Google Play Services yet.  This could actually halt our public rollout of v2.0, as according to the comments on the above gmaps-api-issue developers are seeing a large number of crashes on devices via the Android developer console.  We don't want to roll out v2.0 to a large number of users just to see a ton of crashes due to this problem.

**Comment by barbeau** [comment]
Will anyone watching this issue please go star the root issue on the Google issue tracker at the link below?

https://code.google.com/p/gmaps-api-issues/issues/detail?id=9021

Not sure if this will help push Google to release the fix faster, but it doesn't hurt.

**Comment by barbeau** [comment]
Can anyone who is following this issue please check your device "Settings->Apps", and see which version of Google Play Services your device is running, along with make/model of device?

Looks like mine (LG G4 w/ Android 6.0) finally got bumped to 9.0.83.  The issue was last confirmed on 8.7.03 so I'm hoping that a) this is the major version release Google was referring to in which the issue is resolved, and b) this version is rolling out to most Android devices.

Also, if you were able to originally reproduce the issue it would be awesome if you could test again and let us know if it is indeed resolved.

Google unfortunately muted email replies on the gmaps issue so we have no way of communicating there with users who previously could reproduce the issue.

**Comment by barbeau** [comment]
From this comment on gmaps issue it appears this may be fixed with 9.0.82:
https://code.google.com/p/gmaps-api-issues/issues/detail?id=9021#c129

**Comment by barbeau** [comment]
Another comment on the gmaps issue saying this is fixed, although Google hasn't officially confirmed it yet:
https://code.google.com/p/gmaps-api-issues/issues/detail?id=9021#c133

I'm going to close this out as fixed for now, unless evidence resurfaces that it's still a problem.

**Comment by barbeau** [comment]
Officially confirmed as fixed by Google - from https://code.google.com/p/gmaps-api-issues/issues/detail?id=9021#c135:

>We can confirm that this issue is now fixed in version 9.0.83 of Google Play Services, released in May 2016.
